test/execution_test.exs

```diff
@@ -17,7 +17,7 @@ defmodule ExecutionTest do
     } = Execution.start(%{user_id: "user-1", post_id: "post-123"})
 
     assert is_binary(event_uuid)
-    assert is_integer(event_time)
+    assert is_struct(event_time)
```

Comment:
This will change with the stringification/integer suggested above. Please also assert the type of `elapsed` (would expect integer).

Reply:
this actually asserts if the event time is a datetime (did not find a better matcher for this). Only after the call to `to_record` will this become an ISO8601 date string. Same thing with elapsed (i.e. it does not exist at this point).